### PR TITLE
Code optimizations reported by static code analysis (2019-05-12)

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
#### Summary

SUMMARY: Performance "Code optimizations reported by static code analysis"

#### Purpose of change

Cleanup code while increase performance and fix minor errors.

#### Describe the solution

See commit messages.

### CODE REVIEW

File: src/activity_handlers.cpp [hunk 1]
```diff
@@ -2665,7 +2665,7 @@ void activity_handlers::cracking_do_turn( player_activity *act, player *p )
         item temporary_item( it.type );
         return temporary_item.has_flag( "SAFECRACK" );
     } );
-    if( !( cracking_tool.size() > 0 || p->has_bionic( bionic_id( "bio_ears" ) ) ) ) {
+    if( !( !cracking_tool.empty() || p->has_bionic( bionic_id( "bio_ears" ) ) ) ) {
```
Comment (AMurkin) [inline src/activity_handlers.cpp]:
```suggestion
    if( cracking_tool.empty() && !p->has_bionic( bionic_id( "bio_ears" ) ) ) {
```

**Comment by jbytheway** [comment]
Well, this is an unfortunate coincidence.  Looks like you've hit many of the same changes I made today in #30418.  I resolved some of these differently, and would disagree with some of your resolutions.  You can see my justifications in my commit messages.

**Comment by ZhilkinSerg** [comment]
Huh. I will resolve conflicts after your PR will be merged. I didn't use NOLINT hint, but your approach with it is definitely better.

**Comment by jbytheway** [comment]
Are you also using `clang-tidy` for these changes?

**Comment by ZhilkinSerg** [comment]
Yes, that was clang module in Resharper.

**Comment by ZhilkinSerg** [comment]
Do not merge before #30418.

**Comment by jbytheway** [comment]
If there are other clang-tidy checks which you have already eliminated all the warnings for (and deem to be useful) please feel free to add them to the `.clang-tidy` file now that we have one.

**Pull request merged**